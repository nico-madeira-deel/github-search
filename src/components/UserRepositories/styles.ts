import styled, { css } from 'styled-components'

export const Item = styled.li`
  cursor: pointer;
  list-style: none;

  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    border-radius: ${theme.border.radius};
    color: ${theme.colors.darkBlue};
    margin-right: ${theme.spacings.xsmall};
    margin-bottom: ${theme.spacings.xsmall};
    padding: ${theme.spacings.xsmall} ${theme.spacings.xxsmall};
    transform: scale(1);
    transition: transform ${theme.transitions.fast};
    width: calc((100% / 3) - ${theme.spacings.xsmall});

    &:hover {
      transform: scale(1.025);
      transition: transform ${theme.transitions.fast};
    }

    &:nth-of-type(3n - 0),
    &:last-of-type {
      margin-right: 0;
    }

    h3 {
      font-size: ${theme.font.sizes.medium};
      word-break: break-word;
    }

    p {
      font-size: ${theme.font.sizes.small};
      margin: ${theme.spacings.xsmall} 0;
    }

    a {
      color: ${theme.colors.darkBlue};
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      height: 100%;
    }

    @media (max-width: ${theme.mediaQuery.mobile}px) {
      margin-right: 0;
      width: 100%;
    }
  `}
`

export const Tags = styled.span`
  --spacing-tag: 0.4rem;

  display: block;
  margin-top: var(--spacing-tag);

  ${({ theme }) => css`
    border-radius: ${theme.border.radius};
    font-size: ${theme.font.sizes.xsmall};
    font-weight: ${theme.font.bold};
    padding: var(--spacing-tag);

    &:nth-of-type(1) {
      background-color: ${theme.colors.pink};
    }

    &:nth-of-type(2) {
      background-color: ${theme.colors.lightBlue};
    }

    &:nth-of-type(3) {
      background-color: ${theme.colors.blue};
    }
  `}
`
