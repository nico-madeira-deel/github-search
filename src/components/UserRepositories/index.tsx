import * as S from './styles'
import { UserRepositoriesResponse } from 'models/user'
import Link from 'next/link'

const UserRepositories = ({
  description,
  forks,
  html_url,
  name,
  stargazers_count,
  watchers
}: UserRepositoriesResponse) => {
  return (
    <S.Item>
      <Link href={html_url} passHref={true}>
        <a target="_blank" rel="noopener noreferrer">
          <h3>{name}</h3>
          {description ? (
            <p data-testid="repository-description">{description}</p>
          ) : null}
          <span>
            <S.Tags>Forks: {forks}</S.Tags>
            <S.Tags>Stars: {stargazers_count}</S.Tags>
            <S.Tags>Watchers: {watchers}</S.Tags>
          </span>
        </a>
      </Link>
    </S.Item>
  )
}

export default UserRepositories
