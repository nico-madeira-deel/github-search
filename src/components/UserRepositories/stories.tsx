import { Story, Meta } from '@storybook/react/types-6-0'
import UserRepositories from '.'
import { mockRepository } from 'utils/tests/mock'

export default {
  title: 'UserRepositories',
  component: UserRepositories
} as Meta

export const Basic: Story = (args) => (
  <UserRepositories {...mockRepository} {...args} />
)

Basic.args = mockRepository
