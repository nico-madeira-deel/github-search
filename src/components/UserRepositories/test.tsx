import { screen } from '@testing-library/react'

import UserRepositories from '.'
import { renderWithTheme } from 'utils/tests/helpers'
import { mockRepository } from 'utils/tests/mock'

describe('<UserRepositories />', () => {
  it('should render properly', () => {
    const { container } = renderWithTheme(
      <UserRepositories {...mockRepository} />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should display description if passed by props', () => {
    renderWithTheme(<UserRepositories {...mockRepository} />)

    expect(screen.getByTestId('repository-description')).toBeInTheDocument()
  })
})
