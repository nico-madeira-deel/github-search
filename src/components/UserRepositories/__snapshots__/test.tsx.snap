// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserRepositories /> should render properly 1`] = `
.c0 {
  cursor: pointer;
  list-style: none;
  background-color: #faf9f9;
  border-radius: 0.5rem;
  color: #555b6e;
  margin-right: 1.6rem;
  margin-bottom: 1.6rem;
  padding: 1.6rem 0.8rem;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transition: -webkit-transform .3s ease-in-out;
  -webkit-transition: transform .3s ease-in-out;
  transition: transform .3s ease-in-out;
  width: calc((100% / 3) - 1.6rem);
}

.c0:hover {
  -webkit-transform: scale(1.025);
  -ms-transform: scale(1.025);
  transform: scale(1.025);
  -webkit-transition: -webkit-transform .3s ease-in-out;
  -webkit-transition: transform .3s ease-in-out;
  transition: transform .3s ease-in-out;
}

.c0:nth-of-type(3n - 0),
.c0:last-of-type {
  margin-right: 0;
}

.c0 h3 {
  font-size: 1.6rem;
  word-break: break-word;
}

.c0 p {
  font-size: 1.4rem;
  margin: 1.6rem 0;
}

.c0 a {
  color: #555b6e;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
}

.c1 {
  --spacing-tag: 0.4rem;
  display: block;
  margin-top: var(--spacing-tag);
  border-radius: 0.5rem;
  font-size: 1.2rem;
  font-weight: 700;
  padding: var(--spacing-tag);
}

.c1:nth-of-type(1) {
  background-color: #ffd6ba;
}

.c1:nth-of-type(2) {
  background-color: #bee3db;
}

.c1:nth-of-type(3) {
  background-color: #89b0ae;
}

@media (max-width:480px) {
  .c0 {
    margin-right: 0;
    width: 100%;
  }
}

<li
  class="c0"
>
  <a
    href="https://github.com/facebook/react"
    rel="noopener noreferrer"
    target="_blank"
  >
    <h3>
      React
    </h3>
    <p
      data-testid="repository-description"
    >
      React is an open source JavaScript library used for designing user interfaces.
    </p>
    <span>
      <span
        class="c1"
      >
        Forks: 
        30500
      </span>
      <span
        class="c1"
      >
        Stars: 
        155000
      </span>
      <span
        class="c1"
      >
        Watchers: 
        6700
      </span>
    </span>
  </a>
</li>
`
